ra_aid/llm.py

```
@@ -158,10 +170,10 @@ def create_llm_client(
     # Handle temperature settings
     if is_expert:
         temp_kwargs = {"temperature": 0} if supports_temperature else {}
-    elif temperature is not None and supports_temperature:
+    elif supports_temperature:
+        if temperature is None:
+            raise ValueError(f"Temperature must be provided for model {model_name} which supports temperature")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)